### PR TITLE
Add referral sources index

### PR DESCRIPTION
**Pull request by JohnColvin**
and related patient link

### CODE REVIEW

**Comment by jfriedlaender** [comment]
Is this really what people want? Maybe someone that has requested it could give feedback? cc @hagen

**Comment by hagen** [comment]
At least two people... Referral sources can produce some useful metrics for a clinic when analysed in bulk. Acquiring them in bulk was the issue. Moreover, referral_sources now aligns with other endpoint signatures. Many thanks for the change.

**Comment by jfriedlaender** [comment]
@hagen is this the format you were expecting and was most useful for you?

**Comment by hagen** [comment]
Ideally, each would have it's own `id`, `updated_at`, `created_at`, etc fields. Without these, querying them is not possible. But @JohnColvin did explain there were limitations to this. Understandable. The change means I can now acquire this data in bulk. Even if it does run a full query each time. You will no doubt get requests for filterable fields on this resource. But I'm happy with this change.

**Comment by JohnColvin** [comment]
Serving referral sources as their own resource, even though the data currently sits on the patient table is a step in that direction. One day, we could switch the underlying data to its own table, and add `id` and the other fields, and enable filtering. It's not on my todo list for now, though.

File: sections/referral_sources.md [hunk 1]
```diff
@@ -56,6 +112,11 @@ There are a few possible types of referral sources. Depending on the type, diffe
   "referrer_type": "Contact",
   "subcategory": null,
   "notes": "note written about this referral",
+  "patient": {
+    "links": {
+      "self": "http://local.cliniko.dev:3000/v1/patients/123"
```
Comment (bpinto) [inline sections/referral_sources.md]: Those URLs are not pointing to the production environment.

Comment (JohnColvin) [inline sections/referral_sources.md]: 🤦‍♂️

**Pull request merged**